src/server.c

```diff
@@ -4054,6 +4057,12 @@ uint64_t getCommandFlags(client *c) {
     return cmd_flags;
 }
 
+static int shouldTunnelOnFailure(client *c) {
+    return (!server.cluster_enabled && c->id != CLIENT_ID_AOF && !c->flag.blocked &&
```

Comment:
Should we also add a check to tunnel only `CLIENT_TYPE_NORMAL` type of connection should be skipped from tunneling ?